### ISSUE TITLE
Dialog plugin: Invoking custom component stopped working in 1.1.2

### ISSUE THREAD

**Issue by ghLoser** [quasarframework/quasar#5112]
I'm getting the following warnings:

- [Vue warn]: Error in render: "TypeError: Cannot read property 'getters' of undefined"
- [Vue warn]: Error in mounted hook: "TypeError: Cannot read property 'show' of undefined"

and the Dialog doesn't show.

On closer inspection the component's $refs element ist an empty object and not a function.

Worked perfectly in 1.0,5, updated directly to 1.1.2 and Boom!

**Comment by pdanpdan** [comment]
Please help us help you.

**Comment by GordonBlahut** [comment]
"getters" sounds like you're using Vuex.
  
Are you passing the "parent" property in the dialog options so you can access the store? It was renamed from "root" in between those 2 versions.

**Comment by ghLoser** [comment]
Sorry, no Vuex involved

component code:
```
<template>
  <q-dialog persistent ref="dialog" @hide="onDialogHide">
    <h1>test</h1>
  </q-dialog>
</template>

<script>
export default {
  methods: {
    show() {
      this.$refs.dialog.show()
    },
    hide() {
      this.$refs.dialog.hide()
    },
    onDialogHide() {
      this.$emit('hide')
    },
  },
}
</script>

<style />
```

how it is called:

```
import DialogCrash from 'components/DialogCrash'

Dialog.create({
  component: DialogCrash,
})
```

And, as stated above,  at runtime the component's $refs element ist an empty object and not a function.

Thank you very much.

**Comment by ghLoser** [comment]
Please forget about the 'getters' error, this came from the calling component after the dialog failed.

Originally, the Dialog.create() function was called in a Vuex action. I rewrote this to be called by a method of the parent component, but still getting the empty this.$refs Object in the show() hook.

**Comment by rstoenescu** [comment]
Fixed in "quasar" v1.1.3 (to be released).

**Comment by ghLoser** [comment]
Sorry,

issue still exists in v.1.1.3.

**Comment by pdanpdan** [comment]
Did you add the Dialog plugin to the list of plugins in quasar.conf?

**Comment by ghLoser** [comment]
I stand corrected, 

But the need to refactor all Dialog components of a running & productive PWA due to the changed api is quite painful.

**Comment by rstoenescu** [comment]
API hasn't changed at all. It's "all over the place" in the docs. If you use a Quasar plugin you need to specify it in quasar.conf.js.

**Comment by ghLoser** [comment]
In my PWA (in production since 2.5 months), in several **actions** (_not parent components!_) I'm creating dialogs with custom components like this:

```
export const showMyCustomDialog = (context) => {
    Dialog.create({component: MyCustomComponent })
}
```
Since the 1.0 beta versions up to and including v 1.0.5, MyCustomComponent had access to this.$store without further ado.

This is no longer the case in since I upgraded to v 1.1.2 and today to v.1.1.3. 

Maybe this was not intended to work in the earlier versions, but it worked, and now this.$store is _undefined_. Passing in **this** as parent prop obviously doesn't help in this constellation. Now I'm passing in the actions's context parameter as an additional prop in order to get access to the store, but hopefully there's a less clumsy way.

I didn't change my code, I only upgraded Quasar. 
This is a great product and I absolutely appreciate your work - I just wanted to let you know that something changed.
